src/Radicle/Internal/Effects/Capabilities.hs

```diff
@@ -137,8 +140,46 @@ instance ReadFile IO where
 instance {-# OVERLAPPABLE #-} ReadFile m => ReadFile (Lang m) where
     readFileS = lift . readFileS
 
+class (Monad m) => FileModule m where
+    fileModuleS :: Text -> m (Either Text Text)
+instance FileModule IO where
+    fileModuleS file = do
+        x <- findModule $ toS file
+        case x of
+            Nothing -> pure . Left $ "Could not find file: " <> file
+            Just f  -> pure . Right $ toS f
+instance FileModule (InputT IO) where
+    fileModuleS = lift . fileModuleS
+instance {-# OVERLAPPABLE #-} FileModule m => FileModule (Lang m) where
+    fileModuleS = lift . fileModuleS
+
 putStrLnS :: (Stdout m) => Text -> m ()
 putStrLnS t = putStrS t >> putStrS "\n"
 
 modifyEnvS :: (GetEnv m r, SetEnv m r) => (Env r -> Env r) -> m ()
 modifyEnvS f = getEnvS >>= setEnvS . f
+
+-- | Importing works as follows:
+--
+--    (1) If RADPATH exists, search there
+--    (2) Otherwise search in data-dir
+--    (3) Finally we search in the current directory.
+findModule :: FilePath -> IO (Maybe FilePath)
+findModule file = do
+    mrp <- lookupEnv "RADPATH"
+    radPathDirs <- case mrp of
+            Nothing -> pure <$> getDataDir
+            Just rp -> pure $ splitPath rp
+    let searchPath = radPathDirs ++ ["."]
+    let doFind xs = case xs of
+            [] -> pure Nothing
+            x:xs' -> do
+                exists <- doesFileExist (x </> file)
+                if exists then pure (Just $ x </> file) else doFind xs'
+    doFind searchPath
+  where
+    splitPath p = reverse $ splitPathH p []
```

Comment:
We could use [`splitSearchPath`](https://hackage.haskell.org/package/filepath-1.4.2.1/docs/System-FilePath-Posix.html#v:splitSearchPath)

Reply:
Nice